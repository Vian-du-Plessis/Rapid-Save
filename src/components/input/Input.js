import React, {forwardRef} from 'react';
import styles from './Input.module.scss';


const Input = forwardRef((props, ref) => {
    return (
        <input
            ref={ ref }
            onChange={props.onChange}
            aria-label={props.ariaLabel}
            type={ props.type || 'text' }
            name={ props.name || '' }
            placeholder={ props.placeholder || 'No Label' }
            defaultValue={props.defaultValue}
        />
    );
});

export default Input;