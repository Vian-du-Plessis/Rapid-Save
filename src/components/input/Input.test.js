import Input from './Input';
import TestRenderer from 'react-test-renderer';
import { act, create } from "react-test-renderer";


describe("Test the Input component", () =>{
    test("Create Rendered Input", () => {

        const testBtn = TestRenderer.create(
            <Input
                label='Add expense'
            />
        );
        const btnTestInstance = testBtn.root;
        expect(btnTestInstance.findByType(Input).props.label).toBe["Add expense"];
    });

    test("Input Matches Snapshot", async () => {
        let root;
        await act(() => {
            root = create(            
                <Input
                    label='Add expense'
                />
            );
        });

        expect(root.toJSON()).toMatchSnapshot();

        act(() => {
            root.update(
                <Input
                    label='Add expense'
                />
            );
            expect(root.toJSON()).toMatchSnapshot();
        })
    })
})
