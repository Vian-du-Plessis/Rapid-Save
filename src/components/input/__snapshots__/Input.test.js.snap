// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Input component  Input Matches Snapshot 1`] = `
<input
  name=""
  placeholder="No Label"
  type="text"
/>
`;

exports[`Test the Input component  Input Matches Snapshot 2`] = `
<input
  name=""
  placeholder="No Label"
  type="text"
/>
`;
